### PR TITLE
Remote target deterministic hash develop

### PR DESCRIPTION
**Pull request by ettec**
Fixes https://smartcontract-it.atlassian.net/browse/KS-379

Depends on: https://github.com/smartcontractkit/chainlink-common/pull/675

Signature set sent to the libocr library was assumed to be identical on each workflow node, this is not the case each node can recieve a unique subset of signatures.  In turn this implies the hash of request message sent to the target nodes varies depending on the sender.   The fix to this issue is to exclude the signature set from the message hash. (message hash is still required to form part of the request id to ensure contents of each request match)

### CODE REVIEW

File: core/capabilities/launcher.go [hunk 1]
```diff
@@ -216,6 +216,12 @@ func (w *launcher) addRemoteCapabilities(ctx context.Context, myDON registrysync
 					int(remoteDON.F+1),
 					w.lggr,
 				)
+
+				if c.RemoteTriggerConfig == nil {
```
Comment (cedric-cordenier) [inline core/capabilities/launcher.go]: @ettec We've added a lot of nil checks here with the same code -- can we somehow move it to a single place that does the initialization?

For example inside remote.NewTriggerPublisher, if it receives a nil config, we instantiate one and call ApplyDefaults on it.

We should apply this pattern for all remote config types too.

File: core/services/registrysyncer/syncer.go [hunk 2]
```diff
@@ -165,17 +165,25 @@ func unmarshalCapabilityConfig(data []byte) (capabilities.CapabilityConfiguratio
 		return capabilities.CapabilityConfiguration{}, err
 	}
 
-	var rtc capabilities.RemoteTriggerConfig
+	var remoteTriggerConfig *capabilities.RemoteTriggerConfig
```
Comment (cedric-cordenier) [inline core/services/registrysyncer/syncer.go:168]: Instantiating this as a pointer is part of the problem described above btw :)

File: core/capabilities/launcher.go (hunk 1 above)
Comment (ettec) [reply core/capabilities/launcher.go]: agree that would be better, moved the check into the publisher/subscriber constructors along with the default application and removed from other places (including common)

File: core/capabilities/remote/trigger_subscriber.go [hunk 3]
```diff
@@ -55,11 +55,14 @@ var _ services.Service = &triggerSubscriber{}
...
 const defaultSendChannelBufferSize = 1000
 
-func NewTriggerSubscriber(config capabilities.RemoteTriggerConfig, capInfo commoncap.CapabilityInfo, capDonInfo capabilities.DON, localDonInfo capabilities.DON, dispatcher types.Dispatcher, aggregator types.Aggregator, lggr logger.Logger) *triggerSubscriber {
+func NewTriggerSubscriber(config *capabilities.RemoteTriggerConfig, capInfo commoncap.CapabilityInfo, capDonInfo capabilities.DON, localDonInfo capabilities.DON, dispatcher types.Dispatcher, aggregator types.Aggregator, lggr logger.Logger) *triggerSubscriber {
 	if aggregator == nil {
 		lggr.Warnw("no aggregator provided, using default MODE aggregator", "capabilityId", capInfo.ID)
 		aggregator = NewDefaultModeAggregator(uint32(capDonInfo.F + 1))
 	}
+	if config == nil {
+		config = &capabilities.RemoteTriggerConfig{}
```
Comment (bolekk) [inline core/capabilities/remote/trigger_subscriber.go:65]: Add an Info log line that you're creating an empty config.

**Review by bolekk — COMMENTED** [review]
Looks good but there are unit test failures that don't look like flakes.

**Review by krehermann — APPROVED** [review]

**Review by cedric-cordenier — APPROVED** [review]

**Pull request merged**